src/main/java/pl/edu/wszib/controller/CarController.java

```diff
@@ -1,16 +1,60 @@
 package pl.edu.wszib.controller;
 
-import org.springframework.web.bind.annotation.GetMapping;
-import org.springframework.web.bind.annotation.RequestMapping;
-import org.springframework.web.bind.annotation.RestController;
+import org.springframework.web.bind.annotation.*;
+import pl.edu.wszib.domain.entity.CarType;
 
 @RestController
-@RequestMapping("/car")
+@RequestMapping("/cars")
 public class CarController {
 
-    @RequestMapping("/hello")
+    //TODO: write implementation to all those methods
+
+    @RequestMapping(value = "/{id}")
+    @GetMapping
+    public String getCarById(@PathVariable("id") Integer id) {
+        return "car by id";
+    }
+
+    @RequestMapping(value = "/{brand}")
+    @GetMapping
+    public String getCarByBrand(@PathVariable("brand") String brand) {
+        return "car by brand";
+    }
+
+    @RequestMapping(value = "/brand/{model}")
+    @GetMapping
+    public String getCarByModel(@PathVariable("model") String model) {
+        return "car by model";
+    }
+
+    @RequestMapping(value = "/{enginecapacity}")
     @GetMapping
-    public String carHello() {
-        return "Car controller in use";
+    public String getCarByEngineCapacity(@PathVariable("enginecapacity") String enginecapacity) {
+        return "car by engine capacity";
     }
+
+    @RequestMapping(value = "/{enginetype}")
+    @GetMapping
+    public String getCarByEngineType(@PathVariable("enginetype") String enginetype) {
+        return "car by engine type";
+    }
+
+    @RequestMapping(value = "/{car}")
+    @PutMapping
+    public String addCar(@PathVariable("car") CarType car) {
```

Comment:
jak chcesz tutaj dodać nowy obiekt to lepiej użyc posta, a nie puta. I z urla chcesz pobrać cały obiekt, czyli musialbys w sciezce podac caly obiekt w jsonie. Lepiej zrobic adnotacje @RequestBody("car") CarType car i skrocic mapowanie do "" lub "/"